src/AccelStepperFirmata.cpp

```diff
@@ -0,0 +1,429 @@
+
+      else if (stepCommand == ACCELSTEPPER_STOP) {
+        if (stepper[deviceNum]) {
+          stepper[deviceNum]->stop();
+          isRunning[deviceNum] = false;
+          reportPosition(deviceNum, true);
+        }
+      }
+
+      else if (stepCommand == ACCELSTEPPER_REPORT_POSITION) {
+        if (stepper[deviceNum]) {
+          reportPosition(deviceNum, false);
+        }
+      }
+
+      else if (stepCommand == ACCELSTEPPER_SET_ACCELERATION) {
+
+        float decodedAcceleration = decodeCustomFloat(argv[2], argv[3], argv[4], argv[5]);
+
+        if (stepper[deviceNum]) {
+          /* 
+            <HACK>
+            All firmata instances of accelStepper have an acceleration value. If a user does not
+            want acceleration we just set the acceleration value high enough so
+            that the chosen speed will be realized by the first step of a movement.
+            This simplifies some of the logic in StepperFirmata and gives us more flexibility
+            should an alternative stepper library become available at a future date.
+          */
+          if (decodedAcceleration == 0.0) {
+            stepper[deviceNum]->setAcceleration(MAX_ACCELERATION);
+          } else {
+            stepper[deviceNum]->setAcceleration(decodedAcceleration);
+          }
+          /*
+            </HACK>
+          */
+        }
+
+      }
+
+      else if (stepCommand == ACCELSTEPPER_SET_SPEED) {
+        // Sets the maxSpeed for accelStepper. We do not use setSpeed here because 
+        // all instances of accelStepper that have been created by firmata are
+        // using acceleration. More info about this hack in ACCELSTEPPER_SET_ACCELERATION.
+        float speed = decodeCustomFloat(argv[2], argv[3], argv[4], argv[5]);
+
+        if (stepper[deviceNum]) {
+          stepper[deviceNum]->setMaxSpeed(speed);
+        }
+
+      }
+
+      else if (stepCommand == MULTISTEPPER_CONFIG) {
+        
+        if (!group[deviceNum]) {
+          numGroups++;
+          group[deviceNum] = new MultiStepper();
+        }
+
+        for (byte i = index; i < argc; i++) {
+          byte stepperNumber = argv[i];
+          
+          if (stepper[stepperNumber]) {
+            groupStepperCount[deviceNum]++;
+            group[deviceNum]->addStepper(*stepper[stepperNumber]);
+          }
+        }
+
+        groupIsRunning[deviceNum] = false;
+      }
+
+      else if (stepCommand == MULTISTEPPER_TO) {
+
+        groupIsRunning[deviceNum] = true;
+        long positions[groupStepperCount[deviceNum]];
+        
+        for (byte i = 0, offset = 0; i < groupStepperCount[deviceNum]; i++) {
+          offset = index + (i * 5);
+          positions[i] = decode32BitSignedInteger(argv[offset], argv[offset+1], argv[offset+2], argv[offset+3], argv[offset+4]);
+        }
+
+        group[deviceNum]->moveTo(positions);
+        
+      }
+
+      else if (stepCommand == MULTISTEPPER_STOP) {
+
+        groupIsRunning[deviceNum] = false;
+        
```

Comment:
@dtex what do you think about also calling`reportGroupComplete` here to make the api more consistent with single stepper stop behavior. This would also help eliminate an issue on the client side (firmata.js) where the `multi-stepper-done` event is left unhandled if the user calls `multiStepperStop`.

Reply:
I think making the API more consistent is an excellent thing to do. I'll make that change.